Orso.Arpa.Application/UrlApplication/UrlAddRoleDto.cs

```diff
@@ -29,7 +30,8 @@ public UrlAddRoleDtoValidator()
             RuleFor(d => d.Id)
                 .NotEmpty();
             RuleFor(d => d.RoleId)
-                .NotEmpty();
+                .NotEmpty()
+                .NotEqual(RoleSeedData.Admin.Id);
```

Comment:
Das hier ist für mich keine formale Validierung mehr, sondern eine inhaltliche und gehört damit in die Domain (also in das File, wo der Command validiert wird). Eigentlich müsste imho eher folgendes geprüft werden: Der RoleLevel des angemeldeten Users darf nicht niedriger sein als der RoleLevel der hier zu verknüpfenden Rolle

Reply:
Validierung ist von der Application in die Domain verschoben.